core/scripts/vrfv2plus/testnet/super_scripts.go

```diff
@@ -0,0 +1,286 @@
+	pubBytes, err := hex.DecodeString(*registerKeyUncompressedPubKey)
+	helpers.PanicErr(err)
+	pk, err := crypto.UnmarshalPubkey(pubBytes)
+	helpers.PanicErr(err)
+	var pkBytes []byte
+	if big.NewInt(0).Mod(pk.Y, big.NewInt(2)).Uint64() != 0 {
+		pkBytes = append(pk.X.Bytes(), 1)
+	} else {
+		pkBytes = append(pk.X.Bytes(), 0)
+	}
+	var newPK secp256k1.PublicKey
+	copy(newPK[:], pkBytes)
+
+	compressedPkHex := hexutil.Encode(pkBytes)
+	keyHash, err := newPK.Hash()
+	helpers.PanicErr(err)
+
+	if len(*linkAddress) == 0 {
+		fmt.Println("\nDeploying LINK Token...")
+		address := helpers.DeployLinkToken(e).String()
+		linkAddress = &address
+	}
+
+	if len(*linkEthAddress) == 0 {
+		fmt.Println("\nDeploying LINK/ETH Feed...")
+		address := helpers.DeployLinkEthFeed(e, *linkAddress, fallbackWeiPerUnitLink).String()
+		linkEthAddress = &address
+	}
+
+	fmt.Println("\nDeploying BHS...")
+	bhsContractAddress := deployBHS(e)
+
+	fmt.Println("\nDeploying Batch BHS...")
+	batchBHSAddress := deployBatchBHS(e, bhsContractAddress)
+
+	var coordinatorAddress common.Address
+	fmt.Println("\nDeploying Coordinator...")
+	coordinatorAddress = deployCoordinator(e, *linkAddress, bhsContractAddress.String(), *linkEthAddress)
+
+	coordinator, err := vrf_coordinator_v2plus.NewVRFCoordinatorV2Plus(coordinatorAddress, e.Ec)
+	helpers.PanicErr(err)
+
+	fmt.Println("\nDeploying Batch Coordinator...")
+	batchCoordinatorAddress := deployBatchCoordinatorV2(e, coordinatorAddress)
+
+	fmt.Println("\nSetting Coordinator Config...")
+	setCoordinatorConfig(
+		e,
+		*coordinator,
+		uint16(*minConfs),
+		uint32(*maxGasLimit),
+		uint32(*stalenessSeconds),
+		uint32(*gasAfterPayment),
+		fallbackWeiPerUnitLink,
+		vrf_coordinator_v2plus.VRFCoordinatorV2PlusFeeConfig{
+			FulfillmentFlatFeeLinkPPM: uint32(*flatFeeLinkPPM),
+			FulfillmentFlatFeeEthPPM:  uint32(*flatFeeEthPPM),
+		},
+	)
+
+	fmt.Println("\nConfig set, getting current config from deployed contract...")
+	printCoordinatorConfig(coordinator)
+
+	if len(*registerKeyUncompressedPubKey) > 0 && len(*registerKeyOracleAddress) > 0 {
+		fmt.Println("\nRegistering proving key...")
+		registerCoordinatorProvingKey(e, *coordinator, *registerKeyUncompressedPubKey, *registerKeyOracleAddress)
+
+		fmt.Println("\nProving key registered, getting proving key hashes from deployed contract...")
+		_, _, provingKeyHashes, configErr := coordinator.GetRequestConfig(nil)
+		helpers.PanicErr(configErr)
+		fmt.Println("Key hash registered:", hex.EncodeToString(provingKeyHashes[0][:]))
+	} else {
+		fmt.Println("NOT registering proving key - you must do this eventually in order to fully deploy VRF!")
+	}
+
+	fmt.Println("\nDeploying consumer...")
+	// This creates a new subscription and adds itself (the new consumer) to the subscription
+	consumerAddress, consumer := eoaDeployConsumer(e, coordinatorAddress.String(), *linkAddress, keyHash, *nativePayment)
+
+	// fmt.Println("\nAdding subscription...")
+	// eoaCreateSub(e, *coordinator)
```

Comment:
Gotta keep this in right?

Reply:
The VRF V2+ Consumer seems to create its own subscription and add itself as a consumer to the new subscription.

- VRF V2+ Constructor calling [subscribe method](https://github.com/smartcontractkit/chainlink/blob/22a60fbcad22367cca4d1ec7604b169e3a38108f/contracts/src/v0.8/dev/vrf/testhelpers/VRFV2PlusSingleConsumerExample.sol#L47C9-L47C9)
- [subscribe](https://github.com/smartcontractkit/chainlink/blob/22a60fbcad22367cca4d1ec7604b169e3a38108f/contracts/src/v0.8/dev/vrf/testhelpers/VRFV2PlusSingleConsumerExample.sol#L106-L111) method creating a new subscription and adding the consumer address as its subscriber

Reply:
Fixed this. Migrated to vrf_v2plus_sub_owner